src/components/Signup.js

```diff
@@ -0,0 +1,178 @@
+import React, { Component } from 'react'
+import { NavLink } from 'react-router-dom'
+import axios from 'axios';
+
+
+class Signup extends Component {
+    constructor() {
+        super()
+        this.state = {
+            name: "",
+            email: "",
+            phone: "",
+            work: "",
+            password: "",
+            cpassword: "",
+            loginFail: false,
+            
+        }
+        this.handleName = this.handleName.bind(this)
+        this.handleEmail = this.handleEmail.bind(this)
+        this.handlePhone = this.handlePhone.bind(this)
+        this.handleWork = this.handleWork.bind(this)
+        this.handlePassword = this.handlePassword.bind(this)
+        this.handleCpassword = this.handleCpassword.bind(this)
+        this.onSubmit = this.onSubmit.bind(this)
+    }
+
+    handleName(e) {
+        this.setState({
```

Comment:
Please consuder prevState while updating new value in state.
Please folow same in all the state updation